org.jacoco.core.test/src/org/jacoco/core/data/ExecutionDataReaderWriterTest.java

```diff
@@ -142,7 +142,7 @@ public void testInvalidVersion() throws IOException {
 		buffer.write(ExecutionDataWriter.BLOCK_HEADER);
 		buffer.write(0xC0);
 		buffer.write(0xC0);
-		final char version = ExecutionDataWriter.FORMAT_VERSION - 1;
+		final char version = (char) (ExecutionDataWriter.FORMAT_VERSION - 1);
```

Comment:
There are implicit casts: `char -> int -> subtraction` , in first case value after subtraction is known at compile time and so one more safe implicit cast `int -> char`. In second case value not known and so javac says "possible lossy conversion from int to char".